Sources/Tentacle/JSONExtensions.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  JSONExtensions.swift
+//  Tentacle
+//
+//  Created by Matt Diephouse on 3/10/16.
+//  Copyright © 2016 Matt Diephouse. All rights reserved.
+//
+
+import Foundation
+import Result
+
+internal func decode<T: Decodable>(_ payload: Data) -> Result<T, DecodingError> {
+    return Result { () -> T in
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .formatted(DateFormatter.iso8601)
+        return try decoder.decode(T.self, from: payload)
+    }
+}
+
+internal func decodeList<T: Decodable>(_ payload: Data) -> Result<[T], DecodingError> {
+    return Result { () -> [T] in
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .formatted(DateFormatter.iso8601)
+        return try decoder.decode([T].self, from: payload)
+    }
+}
+
+extension DecodingError: Equatable {
+    static public func ==(lhs: DecodingError, rhs: DecodingError) -> Bool {
+        switch (lhs, rhs) {
+        default: return false // FIXME
```

Comment:
FIXME

Reply:
Any idea how to do proper Equatable implementation for `DecodingError`? 🤔

Reply:
I have a proposition suggested to me in Slack in f9d214e, let me know what you think!